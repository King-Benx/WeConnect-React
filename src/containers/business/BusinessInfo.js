import React from 'react';
import brief_case from '../../images/briefcase.ico';
import { Image } from 'react-bootstrap';

class BusinessInfo extends React.Component {
    constructor(props){
        super(props);
        this.state = {}
    }

    render() { 
        return (
        <div className="col-xs-12">
            <Image src={ brief_case } alt="" thumbnail responsive/>
            <h4><b>Business name: </b> <span className="text text-info">{ this.props.business.name }</span></h4>
            <h4><b>Category: </b> <span className="text text-info">{ this.props.business.category }</span></h4>
            <h4><b>Location: </b> <span className="text text-info">{ this.props.business.location }</span></h4>
            <h4><u>Description</u></h4>
            <p className="text text-info">{ this.props.business.description }</p>
        </div>  )
    }
}
 
export default BusinessInfo;