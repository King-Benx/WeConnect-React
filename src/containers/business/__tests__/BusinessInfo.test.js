import React from 'react';
import { shallow } from 'enzyme';
import BusinessInfo from "../BusinessInfo";
import renderer from 'react-test-renderer';

describe('These are tests for the businesses info page ', () =>{
    const props = {
        business:{
        }
    }

    it('renders without crashing', () => {
       shallow(<BusinessInfo {...props} />)
       const tree = renderer.create(<BusinessInfo {...props} />).toJSON();
       expect(tree).toMatchSnapshot();
    });

    it('has divs', () => {
        const wrapper = shallow(<BusinessInfo {...props}/>)
        expect(wrapper.find('div').length).toBe(1)
    });

    it('has h4s', () => {
        const wrapper = shallow(<BusinessInfo {...props}/>)
        expect(wrapper.find('h4').length).toBe(4)
    });

    it('has b', () => {
        const wrapper = shallow(<BusinessInfo {...props}/>)
        expect(wrapper.find('b').length).toBe(4)
    });

    it('has an underline', () => {
        const wrapper = shallow(<BusinessInfo {...props}/>)
        expect(wrapper.find('u').length).toBe(1)
    });

    it('has an spans', () => {
        const wrapper = shallow(<BusinessInfo {...props}/>)
        expect(wrapper.find('span').length).toBe(3)
    });

    it('has a paragraph', () => {
        const wrapper = shallow(<BusinessInfo {...props}/>)
        expect(wrapper.find('p').length).toBe(1)
    });

    it('has an image', () => {
        const wrapper = shallow(<BusinessInfo {...props}/>)
        expect(wrapper.find('Image').length).toBe(1)
    });

});