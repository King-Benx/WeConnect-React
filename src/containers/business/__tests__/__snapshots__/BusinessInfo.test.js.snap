// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`These are tests for the businesses info page  renders without crashing 1`] = `
<div
  className="col-xs-12"
>
  <img
    alt=""
    className="img-responsive img-thumbnail"
    src="briefcase.ico"
  />
  <h4>
    <b>
      Business name: 
    </b>
     
    <span
      className="text text-info"
    />
  </h4>
  <h4>
    <b>
      Category: 
    </b>
     
    <span
      className="text text-info"
    />
  </h4>
  <h4>
    <b>
      Location: 
    </b>
     
    <span
      className="text text-info"
    />
  </h4>
  <h4>
    <u>
      <b>
        Description
      </b>
    </u>
  </h4>
  <p
    className="text text-info"
  />
</div>
`;
